index.js

```diff
@@ -50,66 +50,61 @@ module.exports = {
   name: require('./package').name,
 
   serverMiddleware: function({ app, options }) {
-    // Calculate livereload settings and cache it to be reused in `contentFor` hook.
-    // Can't do that one in another hook cause it depends on middleware options,
-    // which are only available in `serverMiddleware` hook.
+    // Configuration is not changeable at run-time. Therefore it's safe to not
+    // register the express middleware at all if addon is disabled and
+    // precalculate dynamic values.
+    if (!this._config.enabled) {
+      return;
+    }
+
+    // Need to recalculate the policy if local development server is used to
+    // support live reload, executing tests in development enviroment via
+    // `http://localhost:4200/tests` and reporting CSP violations on CLI.
+    let policyObject = this._config.policy;
+
+    // the local server will never run for production builds, so no danger in
+    // adding the nonce all the time even so it's only needed if tests are
+    // executed by opening `http://localhost:4200/tests`
+    appendSourceList(policyObject, 'script-src', "'nonce-" + STATIC_TEST_NONCE + "'");
```

Comment:
As an extra precaution, perhaps we should check `environment === 'test' || environment === 'development'`?

Reply:
I think the most precise test would be if `tests` is enabled. This seems to be controlled by `tests` property of `EmberApp`, which should be available as `app.tests` in `included` hook and maybe also as `this.app.tests` directly here. Let me double check.

Reply:
This was an awesome hint! :hugs: `app.tests` allows us to always add the nonce required for test support if and only if the build includes tests. This also supports edge cases like `ember test --environment production`.

This relies on the assumption that hopefully nobody deploys a build including tests. But if we don't do that assumption we would break test support for some environments. And this would also not give us that much coincidence cause someone might deploy a development build?!?